config.c

```diff
@@ -246,37 +246,69 @@ static char *readPath(const char *configFile, int lineNum, const char *key,
 /* set *pUid to UID of the given user, return non-zero on failure */
 static int resolveUid(const char *userName, uid_t *pUid)
 {
-    struct passwd *pw;
+    const struct passwd *pw;
+    char *endptr;
+    unsigned long int parsed_uid;
+
 #ifdef __CYGWIN__
     if (strcmp(userName, "root") == 0) {
         *pUid = 0;
         return 0;
     }
 #endif
+
     pw = getpwnam(userName);
-    if (!pw)
-        return -1;
-    *pUid = pw->pw_uid;
-    endpwent();
-    return 0;
+    if (pw) {
+        *pUid = pw->pw_uid;
+        return 0;
+    }
+
+    parsed_uid = strtoul(userName, &endptr, 10);
+    if (userName[0] != '\0' &&
+        *endptr == '\0' &&
+        //parsed_uid != ULONG_MAX &&
```

Comment:
Line comments are not used in logrotate source code.

Reply:
Changed to `/* */`.
The intention of these comment are that we want to check against the error return value of `strtoul` (`ULONG_MAX`), but since we check `< INT_MAX`, its redundant; but in case the code gets refactored at some point one should remember that `ULONG_MAX` signals an error